api/src/test/java/org/openmrs/DrugOrderTest.java

```diff
@@ -314,4 +314,95 @@ public void setAutoExpireDateBasedOnDuration_shouldNotCalculateIfAutoExpireDateA
 		
 		assertEquals(expectedAutoExpireDate, drugOrder.getAutoExpireDate());
 	}
+
+	/**
+	 * @verifies return true if the drugs match and drug non coded match
+	 * @see DrugOrder#hasSameOrderableAs(Order)
+	 */
+	@Test
+	public void hasSameOrderableAs_shouldReturnTrueIfTheConceptsMatchAndHaveSameDrugNonCoded() throws Exception {
+		DrugOrder order = new DrugOrder();
+		Concept concept = new Concept();
+		Drug drug1 = new Drug();
+		drug1.setConcept(concept);
+		order.setDrug(drug1);
+		order.setDrugNonCoded("Chrocine");
+
+		DrugOrder otherOrder = new DrugOrder();
+		otherOrder.setConcept(concept);
+		otherOrder.setDrug(drug1);
+		otherOrder.setDrugNonCoded("chrocine");
+
+		assertTrue(order.hasSameOrderableAs(otherOrder));
+	}
+
+	/**
+	 * @verifies return false if the concepts match and drug non coded do not match
+	 * @see DrugOrder#hasSameOrderableAs(Order)
+	 */
+	@Test
+	public void hasSameOrderableAs_shouldReturnFalseIfTheConceptsMatchAndHaveDifferentDrugNonCoded() throws Exception {
+		DrugOrder order = new DrugOrder();
+		Concept concept = new Concept();
+		order.setDrugNonCoded("Chrocine");
+
+		DrugOrder otherOrder = new DrugOrder();
+		otherOrder.setConcept(concept);
+		otherOrder.setDrugNonCoded("paracetemol");
+
+		assertFalse(order.hasSameOrderableAs(otherOrder));
+	}
+
+	/**
+	 * @verifies return true if the concepts match and drug non coded match trimming spaces
+	 * @see DrugOrder#hasSameOrderableAs(Order)
+	 */
+	@Test
+	public void hasSameOrderableAs_shouldReturnTrueIfTheConceptsMatchAndHaveSameDrugNonCodedTrimmingSpaces() throws Exception {
```

Comment:
That's right, but the acceptance criteria is more correct, the concept should be DRUG OTHER or of class drug